### PR TITLE
node drones hard deletes

### PR DESCRIPTION
**Pull request by Ben10Omintrix**
## About The Pull Request
i noticed node drones were being hard deleted 60% of the time. its because the ore vent wasnt cleaning up after them. also fixes some runtimes if ur id card didnt have a registered account

## Why It's Good For The Game
fix hard deletes

## Changelog
:cl:
/:cl:

### CODE REVIEW

File: code/game/objects/structures/lavaland/ore_vent.dm [hunk 1]
```diff
@@ -284,9 +287,11 @@
 		if(!user_id_card)
 			continue
 		var/point_reward_val = (MINER_POINT_MULTIPLIER * boulder_size) - MINER_POINT_MULTIPLIER // We remove the base value of discovering the vent
-		user_id_card.registered_account.mining_points += point_reward_val
-		user_id_card.registered_account.bank_card_talk("You have been awarded [point_reward_val] mining points for your efforts.")
+		if(user_id_card.registered_account)
```
Comment (JohnFulpWillard) [inline code/game/objects/structures/lavaland/ore_vent.dm:290]: Don't all ID cards have an account? Even just as a placeholder one that can be overriden later?

Comment (Ben10Omintrix) [reply code/game/objects/structures/lavaland/ore_vent.dm:290]: I've no clue just something I noticed while investigating the harddel

Comment (Ben10Omintrix) [reply code/game/objects/structures/lavaland/ore_vent.dm:290]: I saw a lot of other places in code check for account Id so I assumed it's necessary

**Comment by LemonInTheDark** [comment]
he has become too powerful

**Pull request merged**